src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -695,6 +711,52 @@ private void determineUnitsToAttackWith(final List<ProTerritory> prioritizedTerr
       Map<Unit, Set<Territory>> sortedUnitAttackOptions =
           tryToAttackTerritories(prioritizedTerritories, alreadyMovedUnits);
 
+      // Clear bombers
+      for (final Territory t : attackMap.keySet()) {
+        attackMap.get(t).getBombers().clear();
+      }
+
+      // Get all units that have already moved
+      final Set<Unit> alreadyAttackedWithUnits = new HashSet<>();
+      for (final Territory t : attackMap.keySet()) {
+        alreadyAttackedWithUnits.addAll(attackMap.get(t).getUnits());
+        alreadyAttackedWithUnits.addAll(attackMap.get(t).getAmphibAttackMap().keySet());
+      }
+
+      // Check to see if any territories can be bombed
+      final Map<Unit, Set<Territory>> bomberMoveMap = territoryManager.getAttackOptions().getBomberMoveMap();
+      for (final Unit unit : bomberMoveMap.keySet()) {
+        if (alreadyAttackedWithUnits.contains(unit)) {
+          continue;
+        }
+        Optional<Territory> maxBombingTerritory = Optional.empty();
+        int maxBombingScore = 3;
+        for (final Territory t : bomberMoveMap.get(unit)) {
+          final boolean territoryCanBeBombed = t.getUnits().someMatch(Matches.UnitCanProduceUnitsAndCanBeDamaged);
+          if (territoryCanBeBombed && canAirSafelyLandAfterAttack(unit, t)) {
+            final int noAABombingDefense = t.getUnits().someMatch(Matches.UnitIsAAforBombingThisUnitOnly) ? 0 : 1;
+            int maxDamage = 0;
+            final TerritoryAttachment ta = TerritoryAttachment.get(t);
+            if (ta != null) {
+              maxDamage = ta.getProduction();
+            }
+            final int numExistingBombers = attackMap.get(t).getBombers().size();
+            final int remainingDamagePotential = maxDamage - 3 * numExistingBombers;
+            final int bombingScore = (1 + 9 * noAABombingDefense) * remainingDamagePotential;
+            if (bombingScore > maxBombingScore) {
+              maxBombingScore = bombingScore;
+              maxBombingTerritory = Optional.of(t);
+            }
+          }
+        }
+        if (maxBombingTerritory.isPresent()) {
+          final Territory t = maxBombingTerritory.get();
+          attackMap.get(t).getBombers().add(unit);
+          sortedUnitAttackOptions.remove(unit);
+          ProLogger.debug("Add bomber (" + unit + ") to " + t);
```

Comment:
The debug statement is an indicator of manual testing. 'test once, never again'  The lack of test puts no pressure on the code design, and there are some problems:
- long method code smell
- nested for loops 
- nested if statements (within nested for loops, very high cyclomatic complexity)
- lack of abstraction, responsibilities are not delegated